@here/harp-vectortile-datasource/lib/VectorTileDecoder.ts

```diff
@@ -125,7 +102,7 @@ export class VectorTileDataProcessor implements IGeometryProcessor {
             this.m_languages
         );
 
-        dataAdapter.process(data, decodeInfo);
+        this.m_dataAdapter.process(data, decodeInfo, this);
```

Comment:
Not related to your code, but this is a bit weird to have a `this` pointer.

It shows that the `VectorTileDataProcessor` does two things. It is a geometry processor and it also does the wireup between DataAdapter and the DecodedTileEmitter.

I think the IGeometryProcessor should be separated into a separate class from the VectorTileDataProcessor, but this can be done later.

Reply:
Yes, makes sense. But I'd rather leave it for a separate PR.